web/src/click_handlers.js

```diff
@@ -827,6 +827,7 @@ export function initialize() {
         // Allow propagation to close gear menu.
         e.preventDefault();
         dark_theme.enable();
+        message_lists.current.view.update_recipient_bar_background_color();
```

Comment:
This surely needs to be called from `dark_theme.enable`, not from the click handler, so the right thing happens regardless of how your theme changes (say, another browser window could have changed the setting).

Reply:
Added this comment:

    // We cannot update recipient bar color using dark_theme.enable/disable due to
    // it being called before message lists are initialized and the order cannot be changed.
    // Also, since these buttons are only visible for spectators which doen't have events,
    // if theme is changed in a different tab, the theme of this tab remains the same.